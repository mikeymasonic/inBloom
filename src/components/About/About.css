.aboutPage {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;
  color: rgba(40,4,78,1);

  .About {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 80%;
    height: fit-content;
    text-align: justify;
    padding: 1em 1.5em 1em 1.5em;
    margin-bottom: 2em;
    background: linear-gradient(120deg, #d3b1fb, #d6b1fb, #e0b3fb, #edb5fb, #fbb7fa, #fbbaea, #fcbcdb, #fcbfce, #fcc1c3, #fcccc3, #fcd3c5, #fcd6c5);
    box-shadow: 0px 0px 4px 4px #070E0D;
    -webkit-animation: fadein 2s; 
    -moz-animation: fadein 2s; 
    animation: fadein 2s;
    animation: scale 1.5s;
    -webkit-animation: scale 1.5s;
    -moz-animation: scale 2s; 

    a { 
      color: #6109AE;
    }

    a:hover { 
      color: #fe083f;
    }

    h1 { 
      font-size: 2em;
      margin: 0.1em 0 0.5em 0;
    }

    h2 {
      font-size: 2.5em;
      margin: 0;
    }

    div { 
      font-size: 1em;
      font-style: italic;
    }

    p {
        -webkit-animation: fadein 5s; 
        -moz-animation: fadein 5s; 
        animation: fadein 5s;
        margin: 1em 3em 1em 3em;
        font-size: 1.3em;
      }

    .aboutContainer {
      display: flex;
      flex-direction: row;
      justify-content: center;
      align-items: center;
      height: 80%;
      width: 100%;
      margin-bottom: 1em;

      img {
        display: block;
        height: 65vh;
        width: auto;
        border-radius: 10px;
        margin: 0;
        border: 4px outset rgba(40,4,78,1);
        box-shadow: 1px 1px 5px 1px rgba(40,4,78,1);
      }

      .crystalBio {
        font-size: 1.1em;
      }

      .devBio {
        font-size: 1.4em;
      }

      .devPhotos {
        height: 55vh;
      }
    }

    .links img {
      width: 1.8em;
      margin: 0.8em 0.3em 0 0.3em;
      height: auto;
      display: inline-flex;
      text-align: right;
      border-style: none;
    }

    .links img:hover {
      transform: scale(1.1);
    }

    .links img:active {
      transform: scale(0.95);
    }

    .creditsContainer {
      margin: 0.9em 0 1.1em 0;
      font-size: 1.1em;

      .char {
        margin: 0 0.8em 0 0.8em;
      }
    }
  }
}

@media (max-width: 1200px) {
  .aboutContainer {

    .About {
      width: 70%;
      padding: 0.5em 1em 1em 1em;
    }
  }
}

@media (max-width: 667px) {
  .aboutContainer {

    .About {
      width: 85%;
      padding: 0.5em 1em 1em 1em;
      font-size: 0.9em;
    }
  }
}

@keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

/* Firefox < 16 */
@-moz-keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

/* Safari, Chrome and Opera > 12.1 */
@-webkit-keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

@keyframes scale {
  0% {
      transform: scale(0.9);
      -webkit-transform: scale(0.9);
  }
  100% {
      transform: scale(1.0);
      -webkit-transform: scale(1.0);
  }
}