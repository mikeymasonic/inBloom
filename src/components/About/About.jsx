import React from 'react';
import { aboutPage } from '../../utils/data';
import styles from './About.css';

const About = () => {
  return (
    <section className={styles.aboutContainer}>
      <section className={styles.About}>
        <h1>About</h1>
        <p>{aboutPage.description}</p>
      </section>
    </section>
  );
};

export default About;
