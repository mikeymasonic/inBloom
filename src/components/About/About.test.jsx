import React from 'react';
import { render, cleanup } from '@testing-library/react';
import About from './About';

describe('About component', () => {
  afterEach(() => cleanup());
  it('renders About', () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
});
