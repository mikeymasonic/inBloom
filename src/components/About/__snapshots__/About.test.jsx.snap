// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders About 1`] = `
<DocumentFragment>
  <section
    class="whatever"
  >
    <h1>
      about
    </h1>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Omnis quas, quo aspernatur harum assumenda eaque temporibus veritatis ab tempore dolor fugiat dicta eos tempora quasi soluta nulla molestias, libero distinctio. Lorem ipsum dolor sit, amet consectetur adipisicing elit. Iure esse inventore veniam accusamus tempora illo adipisci sequi quaerat consectetur ipsum. Pariatur eveniet delectus a, vel cumque quibusdam autem soluta adipisci? Lorem ipsum dolor sit amet consectetur adipisicing elit. Dolorum quae reprehenderit cumque nulla magnam, sit voluptatibus est laudantium veritatis vel delectus vero adipisci laborum similique doloremque optio! Odio, vero iure!
    </p>
  </section>
</DocumentFragment>
`;
